### ISSUE TITLE
Incremental search missing

### ISSUE THREAD

**Issue by TerDale** [gerib/userscripts#2]
This script would be even more helpful with the addition of a search field, with incremental processing.
What do you think?

**Comment by gerib** [comment]
There is a search field at the top-right originally. Please elaborate what you exactly mean by "incremental processing".

**Comment by TerDale** [comment]
Yes, that's right. Anyway, what I meant is having a search box in the tree-view, and while characters are keyed in, the content of the tree-view is progressively limited to what matches. This is easier for determining if some class/interface does exist based on a few characters. One could do the same with the top RHS box, but it's less handy as not displayed as a tree view.
So, here its is... just an improvement idea ;-)

**Comment by gerib** [comment]
I see. This should be feasible. However, developing a filterable tree in JS myself is a nice challenge but I definitely don't have time for this atm and in the near future. Do you know, by chance, if there is a library that offers such?

**Comment by gerib** [comment]
The problem with searching (or filtering, which I tried first) is that every time one selects a module, package or type lazy-loading is performed now. Loading the DOM trees of _all_ module/package/type pages at once has potential performance and memory impacts and I would have to rewrite the script substantially.

**Comment by gerib** [comment]
Idea for a possible implementation in the future:

Use a (background) [Worker] to collect the infos for _all_ types in the tree view. See [SO: _How can I load a shared web worker with a user-script?_][SO].

[Worker]: https://developer.mozilla.org/en-US/docs/Web/API/Worker
[SO]: https://stackoverflow.com/a/63287809/1744774
